mhs-reference-implementation/mhs/handler/async_response_handler.py

```diff
@@ -0,0 +1,43 @@
+from tornado.web import RequestHandler
+
+import mhs.builder.ebxml_ack_message_builder as ack_builder
+import mhs.builder.ebxml_message_builder as builder
+import mhs.parser.ebxml_message_parser as parser
+from mhs.sender.sender import PARTY_ID
+
+
+class AsyncResponseHandler(RequestHandler):
+    """A RequestHandler for asynchronous responses from a remote MHS."""
+
+    def initialize(self, ack_builder, message_parser):
```

Comment:
just wondering why this constructor isn't `__init__`?

Reply:
It's because `RequestHandler`s are instantiated by Tornado. You pass the class and a list of arguments to pass to the `initialize` method when setting up your application (see [main.py](https://github.com/nhsconnect/integration-adaptors/blob/dab73f2e0e1167b169ecd8a6617237bd777e3779/mhs-reference-implementation/main.py#L34)).

Details are in the [Tornado doc on the `Application` object](https://www.tornadoweb.org/en/stable/guide/structure.html#the-application-object).

Reply:
👍